vortexor/src/stake_updater.rs

```diff
@@ -0,0 +1,101 @@
+//! Module responsible for updating the staked key map.
+//! Adapted from jito-relayer code.
+
+use {
+    crate::rpc_load_balancer::RpcLoadBalancer,
+    log::warn,
+    solana_client::client_error,
+    solana_sdk::pubkey::Pubkey,
+    solana_streamer::streamer::StakedNodes,
+    std::{
+        collections::HashMap,
+        str::FromStr,
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc, RwLock,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::{Duration, Instant},
+    },
+};
+
+// The interval to refresh the stake information.
+// The stakes are updated at the validator every epoch, there is no need to
+// refresh it via RPC too frequently.
+const STAKE_REFRESH_INTERVAL: Duration = Duration::from_secs(1800);
```

Comment:
Not going to block on this - interval is likely simpler.

Is there a reason to refresh on an interval instead of only when there's a new epoch?
We are getting the up-to-date slots of all the rpc we are connected with.
It seems like we don't need to do this every 30 minutes if we just know which epoch we're in, and when it ends. if slot subscription passes the epoch boundary, then get updates stakes.

Reply:
I agreed we can optimize this further. We may get obtain the EpochSchedule to make smarter trigger of the request.

Reply:
opened an issue https://github.com/anza-xyz/agave/issues/5515 to track